s3_storage_client_test.go

```diff
@@ -285,6 +285,23 @@ func TestGetHeaderUnexpectedError(t *testing.T) {
 	}
 }
 
+func TestGetHeaderNonApiError(t *testing.T) {
+	othErr := errors.New("Something went wrong here.")
+	meta := fileMetadata{
+		client:    s3.New(s3.Options{}),
+		stageInfo: &execResponseStageInfo{Location: ""},
+		mockHeader: mockHeaderAPI(func(ctx context.Context, params *s3.HeadObjectInput, optFns ...func(*s3.Options)) (*s3.HeadObjectOutput, error) {
+			return nil, othErr
+		}),
+	}
+	if header, err := new(snowflakeS3Client).getFileHeader(&meta, "file.txt"); header != nil || err == nil {
+		t.Fatalf("expected null header, got: %v", header)
+	}
+	if meta.resStatus != errStatus {
```

Comment:
We have some assertion library - please, check `assertEqualF`.

Reply:
this is part of the original driver code, but will check it

Reply:
ok